app/inject/DonutShopModule.java

```diff
@@ -0,0 +1,60 @@
+package inject;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import io.sphere.sdk.client.PlayJavaSphereClient;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.client.SphereClientFactory;
+import io.sphere.sdk.http.ApacheHttpClientAdapter;
+import org.apache.http.impl.nio.client.HttpAsyncClients;
+import pactas.Pactas;
+import pactas.PactasImpl;
+import play.Configuration;
+import play.Environment;
+import play.Logger;
+import services.*;
+
+import javax.inject.Singleton;
+
+import static java.util.Objects.requireNonNull;
+
+public class DonutShopModule extends AbstractModule {
```

Comment:
You don't close the sphere client when the application are supposed to be shutdown.